src/index.js

```diff
@@ -13,14 +13,35 @@ export function doubleXTimes ( x, num ) {
   return result;
 }
 
-export function doubleEach ( arr ) {
-  let result = arr.map( double );
+export const doubleEach = x => x.map(double);
 
-  // const result = arr.map( x => double( x ) );
-  // const result = arr.map( function ( x ) {
-  //   return double( x );
-  // });
+export function Animal() {
+    this.clorophyll = false;
+}
+Animal.SOUND = 'generic sound';
+Animal.prototype.speak = function () {return this.constructor.SOUND};
 
-  return result;
+export function Reptile() {
+    this.scales = true;
+}
+
+Reptile.SOUND = 'Roar';
+Reptile.prototype = new Animal();
+Reptile.prototype.constructor = Reptile;
+
+export function Primate() {
+    this.thumbs = true;
```

Comment:
You'll want to call the `Animal` constructor within this constructor, and ["bind" it to this instance](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call).